packages/client/modules/userDashboard/components/OrgBilling/NewBillingLeaderInput.tsx

```diff
@@ -0,0 +1,147 @@
+import styled from '@emotion/styled'
+import {Person} from '@mui/icons-material'
+import graphql from 'babel-plugin-relay/macro'
+import React, {ChangeEvent, FormEvent, useEffect, useRef} from 'react'
+import {useFragment} from 'react-relay'
+import {MenuPosition} from '~/hooks/useCoords'
+import useMenu from '~/hooks/useMenu'
+import {PALETTE} from '~/styles/paletteV3'
+import {NewBillingLeaderInput_organization$key} from '~/__generated__/NewBillingLeaderInput_organization.graphql'
+import StyledError from '../../../../components/StyledError'
+import useForm from '../../../../hooks/useForm'
+import {PortalStatus} from '../../../../hooks/usePortal'
+import NewBillingLeaderMenu from './NewBillingLeaderMenu'
+import {ElementWidth} from '../../../../types/constEnums'
+
+const StyledIcon = styled(Person)({
+  color: PALETTE.SKY_500,
+  width: ElementWidth.BILLING_AVATAR,
+  height: ElementWidth.BILLING_AVATAR
+})
+
+const Form = styled('form')({
+  display: 'flex',
+  flexDirection: 'column',
+  width: '100%'
+})
+
+const Item = styled('div')({
+  backgroundColor: PALETTE.SLATE_200,
+  cursor: 'pointer',
+  display: 'flex',
+  alignItems: 'center',
+  width: '100%'
+})
+
+const Wrapper = styled('div')({
+  display: 'flex',
+  flexDirection: 'column',
+  paddingLeft: 16,
+  width: '100%'
+})
+
+const DropdownWrapper = styled('button')({
+  width: 400
+})
+
+const NewLeaderInput = styled('input')({
+  appearance: 'none',
+  background: 'transparent',
+  border: 'none',
+  color: PALETTE.SLATE_700,
+  fontSize: 16,
+  margin: 0,
+  paddingRight: 8,
+  outline: 0,
+  width: '100%'
+})
+
+const Error = styled(StyledError)({
+  fontSize: 13,
+  textAlign: 'left',
+  width: '100%'
+})
+
+interface Props {
+  organizationRef: NewBillingLeaderInput_organization$key
+  removeInput: () => void
+}
+
+const NewBillingLeaderInput = (props: Props) => {
+  const {removeInput, organizationRef} = props
+  const organization = useFragment(
+    graphql`
+      fragment NewBillingLeaderInput_organization on Organization {
+        ...NewBillingLeaderMenu_organization
+      }
+    `,
+    organizationRef
+  )
+  const {fields, onChange} = useForm({
+    newLeader: {
+      getDefault: () => ''
+    }
+  })
+  const {dirty, error, value: newLeaderValue} = fields.newLeader
+  const ref = useRef<HTMLInputElement>(null)
+
+  const {originRef, menuPortal, menuProps, togglePortal, portalStatus} = useMenu(
+    MenuPosition.UPPER_CENTER,
+    {isDropdown: true}
+  )
+  useEffect(() => {
+    if (portalStatus === PortalStatus.Exited) {
+      ref.current?.focus()
+    }
+  }, [portalStatus])
+
+  const handleCreateNewLeader = (e: FormEvent) => {
+    e.preventDefault()
```

Comment:
-1 can you please add a comment why this is necessary here? It makes it difficult to refactor in the future otherwise.